test/widgets/action_sheet_test.dart

```diff
@@ -85,6 +90,79 @@ void main() {
     (store.connection as FakeApiConnection).prepare(httpStatus: 400, json: fakeResponseJson);
   }
 
+  group('ShareButton', () {
+    // Tests should call setupMockSharePlus.
+    setUp(() async {
+      TestZulipBinding.ensureInitialized();
+      TestWidgetsFlutterBinding.ensureInitialized();
+    });
+
+    MockSharePlus setupMockSharePlus() {
+      final mock = MockSharePlus();
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+        MethodChannelShare.channel,
+        mock.handleMethodCall,
+      );
+      return mock;
+    }
+
+    tearDown(() async {
+      TestZulipBinding.instance.reset();
+    });
+
+    Future<void> tapShareButton(WidgetTester tester) async {
+      await tester.ensureVisible(find.byIcon(Icons.adaptive.share, skipOffstage: false));
+      await tester.tap(find.byIcon(Icons.adaptive.share));
+      await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
+    }
+
+    testWidgets('request succeeds; sharing succeeds', (WidgetTester tester) async {
+      final mockSharePlus = setupMockSharePlus();
+      final message = eg.streamMessage();
+      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+      final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
```

Comment:
```suggestion
      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
```

(this may be another fruit of rebasing)